Mathlib/Data/Finsupp/Basic.lean

```diff
@@ -214,6 +214,11 @@ def mapRange.addMonoidHom (f : M →+ N) : (α →₀ M) →+ α →₀ N
   map_add' a b := by dsimp only; exact mapRange_add f.map_add _ _; -- Porting note: `dsimp` needed
 #align finsupp.map_range.add_monoid_hom Finsupp.mapRange.addMonoidHom
 
+lemma mapRange.addMonoidHom_apply_single
+    [AddCommMonoid N] [AddCommMonoid P] (e : N →+ P) (a : α) (n : N) :
+    mapRange.addMonoidHom e (single a n) = single a (e n) := by
+    simp only [addMonoidHom_apply, mapRange_single]
+
```

Comment:
I think this is best kept as separate lemmas; repeating every lemma about a function `foo` for `fooOneHom`, `fooZeroHom` etc doesn't scale.

Reply:
I'm sorry, I don't understand what you mean in the first part of the sentence. 
@erdOne suggested to make it work for `AddHomClass` in #12293, and that works.
(I agree about the second part!)

Reply:
At least, one can give the shorter (but less general) definition 
```
/-- RingHom functoriality for the monoid algebra -/
noncomputable def ringHom [Monoid α] (e : N →+* P) :
    MonoidAlgebra N α →+* MonoidAlgebra P α := by
  apply liftNCRingHom (singleOneRingHom.comp e) (of P α)
  intro n a
  simp only [commute_iff_eq, RingHom.coe_comp, Function.comp_apply,
    singleOneRingHom_apply, ZeroHom.toFun_eq_coe, AddMonoidHom.toZeroHom_coe,
    singleAddHom_apply, of_apply, single_mul_single, one_mul, mul_one]
```
and I can try proving the lemmas for them. (This is not as obvious, I need to understand why!)

Reply:
(That works, but one loses the fact that it corresponds with docs#Finsupp.mapRange…)

Reply:
I mean that the caller should be expected to write `rw [addMonoidHom_apply, mapRange_single]` rather than using this lemma; or `dsimp; rw [mapRange_single]`. The first step in most proofs should be to let `dsimp` clean up any bundled functions to bring things into a standard form that we have lemmas for.

Reply:
I deleted the lemma.